src/core.js

```diff
@@ -848,7 +848,7 @@ $.extend( $.validator, {
 			// aria-describedby should directly reference the error element
 			if ( describer ) {
 				selector = selector + ", #" + describer
-					.replace( /'/g, "\\'" )
+					.replace( /('|:|\.|\[|\]|\$)/g, "\\$1" )
```

Comment:
Would it make sense to extract this pattern into a var so this and line 814 cannot get out of sync?